lib/json_logic/operation.rb

```diff
@@ -128,10 +143,47 @@ def self.is_iterable?(operator)
       ['filter', 'some', 'all', 'none', 'in', 'map', 'reduce'].include?(operator.to_s)
     end
 
+    def self.is_nilable?(operator)
+      [
+        'substr',
+        '=',
+        '==',
+        '===',
+        '!=',
+        '!==',
+        '>',
+        '>=',
+        '<',
+        '<=',
+        'max',
+        'min',
+        '+',
+        '-',
+        '*',
+        '/',
+        '%',
+        '^'
+      ].include?(operator.to_s)
+    end
+
     def self.add_operation(operator, function)
       self.class.send(:define_method, operator) do |v, d|
         function.call(v, d)
       end
     end
+
+    def self.format_values(values)
+      values = Array(values).flatten
+      # If at least 1 value is numeric, assume all values can be treated as such
+      # Sometimes numbers are represented as strings, so they need to be converted
+      if values.any? { |v| v.is_a?(Numeric) }
+        values.map(&:to_f)
+      elsif values.all? { |v| v.is_a?(String) }
+        values.map(&:downcase)
```

Comment:
might be nice if the cases of `downcase` are tested

Reply:
none of the default operators will actually hit this case, but we need this for the custom operators we have in the web repo. those already have specs that test this

Reply:
I think porting a bare minimum of specs that test `downcase` would be good -- would make it easier to make changes to this repo and not break stuff!

Reply:
ok added some tests for format_values directly https://github.com/persona-id/json-logic-ruby/pull/4